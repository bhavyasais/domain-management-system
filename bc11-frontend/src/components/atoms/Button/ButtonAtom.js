import React from "react";
import Button from "@material-ui/core/Button";
import { makeStyles, Typography } from "@material-ui/core";
import PropTypes from "prop-types";
import clsx from "clsx";
const useStyles = makeStyles({
  buttonLabel: {
    overflow: "hidden",
    textOverflow: "ellipsis",
  },
});

export default function ButtonAtom({
  value,
  variant,
  disabled,
  color,
  onClick,
  textColor,
  rootStyle,
  labelStyle,
  textVariant,
  maxWidth,
  ...restProps
}) {
  const classes = useStyles();
  return (
    <div>
      <Button
        variant={variant}
        disabled={disabled}
        style={{ backgroundColor: color, maxWidth: maxWidth }}
        onClick={onClick}
        classes={{
          root: rootStyle,
          label: clsx(labelStyle, classes.buttonLabel),
        }}
        {...restProps}
      >
        <Typography variant={textVariant} className={classes.buttonLabel}>
          {value}
        </Typography>
      </Button>
    </div>
  );
}
Button.propTypes = {
  value: PropTypes.string,
  color: PropTypes.string,
  variant: PropTypes.string,
};
Button.defaultProps = {
  variant: "contained",
};
