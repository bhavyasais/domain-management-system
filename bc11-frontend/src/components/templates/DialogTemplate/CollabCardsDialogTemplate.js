/* eslint-disable import/prefer-default-export */
import React, { useState, useEffect, useCallback } from 'react';
import Grid from '@material-ui/core/Grid';
import { withStyles, makeStyles } from '@material-ui/core/styles';
import Dialog from '@material-ui/core/Dialog';
import MuiDialogTitle from '@material-ui/core/DialogTitle';
import Typography from '@material-ui/core/Typography';
import MuiDialogContent from '@material-ui/core/DialogContent';
import IconButtonAtom from '../../atoms/Icon/IconAtom';
import IconAtom from '../../atoms/Icon/IconAtom';
import options from '../../atoms/shared/ListIcons';
import CollaboratorService from '../../../services/CollaboratorService';
import CollabCard from '../../molecules/CollabCard/CollabCard';
import GoogleSuiteService from '../../../services/GoogleSuiteService';

const useStyles = makeStyles({
  root: {
    flexGrow: 1,
    minWidth: '926px',
    width: '926px',
    height: '544px',
  },
});

const styles = (theme) => ({
  root: {
    margin: 0,
    padding: theme.spacing(3),
  },
  closeButton: {
    position: 'absolute',
    right: theme.spacing(1),
    top: theme.spacing(1),
    color: theme.palette.grey[500],
  },
});

const DialogTitle = withStyles(styles)((props) => {
  const { children, classes, onClose, ...other } = props;
  return (
    <MuiDialogTitle disableTypography className={classes.root} {...other}>
      <Typography variant="h6">{children}</Typography>
      {onClose ? (
        <IconButtonAtom aria-label="close" className={classes.closeButton} onClick={onClose}>
          <IconAtom children={options.close} />
        </IconButtonAtom>
      ) : null}
    </MuiDialogTitle>
  );
});

const DialogContent = withStyles((theme) => ({
  root: {
    padding: theme.spacing(2),
  },
}))(MuiDialogContent);

export const DialogTemplate = ({ handleClose, open, onClick }) => {
  const [collaborations, setCollaborations] = useState([]);

  const handleGoogleSignIn = async (item) => {
    try {
      const GoogleAuth = await window.gapi.auth2.getAuthInstance();
      const result = await GoogleAuth.grantOfflineAccess();
      console.log('result after login', result.code);
      const params = new URLSearchParams();
      params.append('authCode', result?.code);
      await GoogleSuiteService.updateToken(params);
      localStorage.setItem('Collaboration', item.id);
      onClick(item);
    } catch (error) {
      console.warn('Error in gogle sign in', error);
    }
  };

  const handleOnClick = (item) => async () => {
    await handleGoogleSignIn(item);
  };
  const classes1 = useStyles();

  const fetchCollaborations = useCallback(async () => {
    try {
      const result = await CollaboratorService.getAllCollaborators();
      setCollaborations(result);
    } catch (error) {
      console.warn('error in fetching departments : ', error);
    }
  }, []);
  useEffect(() => {
    fetchCollaborations();
  }, []);

  return (
    <div className={classes1.root}>
      <Dialog
        onClose={handleClose}
        aria-labelledby="customized-dialog-title"
        open={open}
        maxWidth="md"
      >
        <DialogTitle id="customized-dialog-title" onClose={handleClose}>
          Choose Collaboration Platforms
        </DialogTitle>
        <DialogContent dividers>
          <Grid container spacing={4}>
            {collaborations.map((item, idx) => (
              <Grid item key={idx}>
                <CollabCard
                  title={item.name}
                  image={item.imageURL}
                  component="img"
                  height={item.height}
                  name={item.name}
                  onClick={handleOnClick(item)}
                />
              </Grid>
            ))}
            ,
          </Grid>
        </DialogContent>
      </Dialog>
    </div>
  );
};
