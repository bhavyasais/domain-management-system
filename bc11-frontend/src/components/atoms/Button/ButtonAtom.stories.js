import React from "react";
import { storiesOf } from "@storybook/react";
import ButtonAtom from "./ButtonAtom";
import { text ,boolean} from "@storybook/addon-knobs";

storiesOf("Atoms/Button", module).add("Contained Button", () => (
  <ButtonAtom
    variant={text("Button Variant","contained")}
    value={text("Text", "Add")}
    // color={color("Color","#1d4cd7")}
    //textVariant={text("Typography Variant","h4")}
    onClick={() => console.log("Button clicked")}
    disabled={boolean('Disabled', false)}
    maxWidth="100px"
    disableRipple
  />));
