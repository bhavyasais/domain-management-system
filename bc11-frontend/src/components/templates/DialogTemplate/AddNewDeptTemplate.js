import React, { useState } from "react";
import TextField from "@material-ui/core/TextField";
import Dialog from "@material-ui/core/Dialog";
import DialogActions from "@material-ui/core/DialogActions";
import DialogContent from "@material-ui/core/DialogContent";
import DialogTitle from "@material-ui/core/DialogTitle";
import ButtonAtom from "../../atoms/Button/ButtonAtom";
import { Typography } from "@material-ui/core";

export default function AddNewDeptTemplate(props) {

    const handleAddClick = () => {
        props.onAddClick(department);
        props.handleClose();
    }

    const [department,setDepartment]= useState('');
    const handleDepartmentChange = (e) => {
        setDepartment(e.target.value);
    }


  return (
    <div>
      <Dialog
        open={props.open}
        onClose={props.handleClose}
        aria-labelledby="form-dialog-title"
      >
        <DialogTitle id="form-dialog-title"><Typography variant="h6">Add New Department</Typography></DialogTitle>
        <DialogContent>
          <form>
            <div>
              <TextField
                id="outlined-basic"
                label="Department"
                variant="outlined"
                onChange={handleDepartmentChange}
              />
            </div>
          </form>
        </DialogContent>
        <DialogActions>
        <ButtonAtom
          
          variant="outlined"
          value="Close"
          textVariant="caption"
          onClick={props.handleClose}
        />
        <ButtonAtom
            variant="contained"
            value="Add"
            textVariant="caption"
            onClick={handleAddClick}
          />
        
           
        </DialogActions>
      </Dialog>
    </div>
  );
}
