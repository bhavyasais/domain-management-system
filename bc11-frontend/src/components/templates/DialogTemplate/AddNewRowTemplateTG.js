import React, { useState } from "react";
import TextField from "@material-ui/core/TextField";
import Dialog from "@material-ui/core/Dialog";
import DialogActions from "@material-ui/core/DialogActions";
import DialogContent from "@material-ui/core/DialogContent";
import DialogTitle from "@material-ui/core/DialogTitle";
import ButtonAtom from "../../atoms/Button/ButtonAtom";
import { Typography } from "@material-ui/core";

export default function AddNewRowTemplateTG(props) {

    const handleAddClick = () => {
        props.onAddClick({name,trust_score,relationship});
        props.handleClose();
    }

   
    const [name,setDomain]= useState('');
    const [trust_score,setTrustScore]= useState('');
    const [relationship,setRelationship]= useState('');

    
    const handleDomainChange = (e) => {
        setDomain(e.target.value);
    }

    const handleTrustScoreChange = (e) => {
        setTrustScore(e.target.value);
    }

    const handleRelationshipChange = (e) => {
      setRelationship(e.target.value);
    }


  return (
    <div>
      <Dialog
        open={props.open}
        onClose={props.handleClose}
        aria-labelledby="form-dialog-title"
      >
        <DialogTitle id="form-dialog-title"><Typography variant="h6">Add New Row</Typography></DialogTitle>
        <DialogContent>
          <form>
          
            <div>
              {" "}
              <TextField
                id="outlined-basic"
                label="Domain"
                variant="outlined"
                onChange={handleDomainChange}
              />
            </div>
            <div>
              {" "}
              <TextField
                id="outlined-basic"
                label="Trust Score"
                variant="outlined"
                onChange={handleTrustScoreChange}
              />
            </div>
            <div>
              {" "}
              <TextField
                id="outlined-basic"
                label="Relationship"
                variant="outlined"
                onChange={handleRelationshipChange}
              />
            </div>
          </form>
        </DialogContent>
        <DialogActions>
        <ButtonAtom
          
          variant="outlined"
          value="Close"
          textVariant="caption"
          onClick={props.handleClose}
        />
        <ButtonAtom
            variant="contained"
            value="Add"
            textVariant="caption"
            onClick={handleAddClick}
          />
        
           
           
        </DialogActions>
      </Dialog>
    </div>
  );
}