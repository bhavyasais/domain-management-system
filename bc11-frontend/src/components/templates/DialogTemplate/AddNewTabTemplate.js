import React, { useState } from 'react';
import TextField from '@material-ui/core/TextField';
import Dialog from '@material-ui/core/Dialog';
import DialogActions from '@material-ui/core/DialogActions';
import DialogContent from '@material-ui/core/DialogContent';
import ButtonAtom from '../../atoms/Button/ButtonAtom';
import { Typography } from '@material-ui/core';

export default function AddNewTabTemplate(props) {
  const handleAddClick = () => {
    props.onAddClick(tabName);
    props.handleClose();
  };

  const [tabName, setTabName] = useState('');
  const handleTabChange = (e) => {
    setTabName(e.target.value);
  };

  return (
    <div>
      <Dialog open={props.open} onClose={props.handleClose} aria-labelledby="form-dialog-title">
        <Typography variant="h6">Add New Tab</Typography>
        <DialogContent>
          <form>
            <div>
              <TextField
                id="outlined-basic"
                label="Tab Name"
                variant="outlined"
                onChange={handleTabChange}
              />
            </div>
          </form>
        </DialogContent>
        <DialogActions>
          <ButtonAtom
            variant="outlined"
            value="Close"
            textVariant="caption"
            onClick={props.handleClose}
          />
          <ButtonAtom
            variant="contained"
            value="Add"
            textVariant="caption"
            onClick={handleAddClick}
          />
        </DialogActions>
      </Dialog>
    </div>
  );
}
