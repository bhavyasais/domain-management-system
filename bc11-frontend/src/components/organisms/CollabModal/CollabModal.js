import React from "react";
import ButtonAtom from "../../atoms/Button/ButtonAtom";
import { DialogTemplate } from "../../templates/DialogTemplate/CollabCardsDialogTemplate";

export default function CollabModal(props) {
  const [open, setOpen] = React.useState(false);

  const handleClickOpen = () => {
    setOpen(true);
  };
  const handleClose = () => {
    setOpen(false);
  };
  const handleOnClick=()=>{
    props.onClick();
  }
  return (
    <div>
      <ButtonAtom
        variant="contained"
        color="primary"
        onClick={handleClickOpen}
        value="Enable Access"
      />
      <DialogTemplate handleClose={handleClose} open={open} onClick={handleOnClick}/>
    </div>
  );
}
