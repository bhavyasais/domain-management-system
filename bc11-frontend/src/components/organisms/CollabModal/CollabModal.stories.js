import React from "react";
import CollabModal from "./CollabModal"; // This is our component
import { storiesOf } from "@storybook/react";
import CloseIcon from "@material-ui/icons/Close";
import options from "../../atoms/shared/ListIcons";

const icons={
  close : <CloseIcon/>,
};
storiesOf("Organisms/CollabModal", module).add("Collab Modal", () => (
  <CollabModal icon={options.close}/>
));
