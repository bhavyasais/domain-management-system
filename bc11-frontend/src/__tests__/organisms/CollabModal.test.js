import CollabModal from '../../components/organisms/CollabModal/CollabModal';
import React from 'react';
import { act } from 'react-dom/test-utils';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import Button from '../../components/atoms/Button/ButtonAtom';
import { DialogTemplate } from '../../components/templates/DialogTemplate/CollabCardsDialogTemplate';
import ButtonAtom from '../../components/atoms/Button/ButtonAtom';

describe('CollabModal', () => {
  let component;
  const handleClick = jest.fn();
  beforeEach(() => {
    act(() => {
      component = mount(<CollabModal />);
    });
  });
  it('should render correctly', () => {
    let wrapper;
    wrapper = renderer.create(<CollabModal />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render one button', () => {
    expect(component.find(Button)).toHaveLength(1);
  });
  it('renders Modal component', () => {
    expect(component.find(DialogTemplate).length).toEqual(1);
  });

  it("opens the modal", () => {
    const wrapper = mount(<CollabModal onClick={handleClick}/>);
    wrapper.find('ButtonAtom').at(0).props().onClick();
    wrapper.find('DialogTemplate').at(0).props().handleClose();
    wrapper.find('DialogTemplate').at(0).props().onClick();
    expect(handleClick).toHaveBeenCalled();
  });


});
