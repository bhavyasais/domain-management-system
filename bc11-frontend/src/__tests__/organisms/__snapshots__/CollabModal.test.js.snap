// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollabModal should render correctly 1`] = `
<div>
  <div>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "backgroundColor": "primary",
          "maxWidth": undefined,
        }
      }
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label makeStyles-buttonLabel-1"
      >
        <p
          className="MuiTypography-root makeStyles-buttonLabel-1 MuiTypography-body1"
        >
          Enable Access
        </p>
      </span>
    </button>
  </div>
  <div
    className="makeStyles-root-2"
  />
</div>
`;
